tulip/abstract/find_controller.py

```diff
@@ -279,10 +279,30 @@ def get_input(
                 ] += mid_weight * np.eye(n)
 
                 r[idx, 0] += -mid_weight * xc
-                u, cost = get_input_helper(
-                    x0, ssys, P1, P3, N, R, r, Q, ord,
-                    closed_loop=closed_loop, solver=solver
-                )
+                try:
+                    u, cost = get_input_helper(
+                        x0, ssys, P1, P3, N, R, r, Q, ord,
+                        closed_loop=closed_loop, solver=solver
+                    )
+                except Exception as ex:
```

Comment:
Catching `Exception` would include all exceptions, also those due to reasons other than unreachability of the target polytope (for example [here](https://github.com/tulip-control/tulip-control/pull/242/files#diff-db73a44c4631fa5fc7c04bff0f7d729aR445-R447)).

The function `get_input_helper` raises `Exception` when the QP solution status is not `'optimal'` ([here](https://github.com/tulip-control/tulip-control/pull/242/files#diff-db73a44c4631fa5fc7c04bff0f7d729aR450-R453)) ([documentation of the `cvxopt` function `qp`](https://cvxopt.org/userguide/coneprog.html#cvxopt.solvers.qp)) and when the LP solution status is not 0 ([here](https://github.com/tulip-control/tulip-control/pull/242/files#diff-db73a44c4631fa5fc7c04bff0f7d729aR475-R478)) ([cases of solution status when using the `cvxopt` function `lp`](https://github.com/tulip-control/polytope/blob/ccb080dbae7ffbb46e2ebe016665d9d0895fc1cd/polytope/solvers.py#L111-L118) and [documentation of the function `scipy.optimize.linprog`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.linprog.html)).

A possibility could be to subclass [`Exception`](https://docs.python.org/3/library/exceptions.html#Exception), for example:

```python
class _GetInputHelperError(Exception):
    """Could not solve the given problem."""
    pass
```

and to `raise _GetInputHelperError` from within the function `get_input_helper` to signify errors in the solution status. This would enable catching `_GetInputHelperError` here, and raising other errors.